### PR TITLE
Feat: rabbit death

### PR DESCRIPTION
**Pull request by fjebaker**
Rabbit loses 1 health every 60 frames (1 second), and starts with 120 health. At 60 health remaining the rabbit's ear droops and it's leg bleeds to indicate it is hurt.

Each item given to the rabbit restores 20 health.

### CODE REVIEW

File: cart/player.go [hunk 1]
```diff
@@ -128,7 +128,12 @@ func (player *Player) HandleInteraction(t int32) {
 	player.Move_fx.Stop()
 }
 
-func (player *Player) Update(t int32, world *World, game *Game, desired *RetrievableItem) {
+func (player *Player) Update(t int32, world *World, game *Game, desired *RetrievableItem, rabbit *Rabbit) {
+	if rabbit.IsDead() && rabbit.PointInZone(player.X, player.Y) {
+		game.ChangeState(GAME_STATE_OVER)
+		player.SetDead()
+	}
```
Comment (jwiggins) [inline cart/player.go]: Touching the dead rabbit kills you, but the rabbit dying is OK?

Actually, I think this makes sense.

File: cart/player.go [hunk 2]
```diff
@@ -202,12 +207,8 @@ func (player *Player) animate(t int32) {
...
 
 func (player *Player) incrementFrame() {
-	if !player.Dead {
-		player.Frame = (player.Frame + 1) % 5
-		player.Sprite.Id = PLAYER_MAIN_FRAME + player.Frame
-	} else {
-		player.Sprite.Id = PLAYER_DEAD_FRAME
-	}
+	player.Frame = (player.Frame + 1) % 5
+	player.Sprite.Id = PLAYER_MAIN_FRAME + player.Frame
```
Comment (jwiggins) [inline cart/player.go:215]: Won't this cause the play to come back to life, at least as far as the sprite is concerned?

File: cart/retrievable-item.go [hunk 3]
```diff
@@ -53,16 +54,18 @@ func (item *RetrievableItem) Id() int32 {
...
 	is_in_zone := rabbit.PointInZone(player.X, player.Y)
+	is_dead := rabbit.IsDead()
 
-	if player.HasItem && is_in_zone {
+	if player.HasItem && is_in_zone && !is_dead {
 		player.HasItem = false
 		item.Sprite.Id = int32(newDesiredItem())
 		rabbit.HappySfx.Play()
 		rabbit.ShowHeart = true
+		rabbit.Health = rabbit.Health + ITEM_HEALING_AMOUNT
```
Comment (jwiggins) [inline cart/retrievable-item.go]: This should probably be a method for the Rabbit.

**Review by jwiggins — COMMENTED** [review]
I think there might be a bug in the player sprite animation logic

File: cart/player.go (hunk 2 above)
Comment (jwiggins) [reply cart/player.go:215]: It works, but I don't know why

Comment (jwiggins) [reply cart/player.go:215]: Ahhh. OK. We set the game to over when the player dies and that stops updates.

File: cart/player.go (hunk 1 above)
Comment (jwiggins) [reply cart/player.go]: You should return immediately after dying. Otherwise the remainder of the update could put the player into a weird state.

Comment (fjebaker) [reply cart/player.go]: Yes true, now that we have a `SetDead` function we don't have to wait for the call to animate to update the sprite.

File: cart/player.go (hunk 2 above)
Comment (fjebaker) [reply cart/player.go:215]: 👍

File: cart/player.go [hunk 4]
```diff
@@ -150,7 +157,7 @@ func (player *Player) Update(t int32, world *World, game *Game, desired *Retriev
 
 	if world.IsDeadly(tileIndex) {
 		game.ChangeState(GAME_STATE_OVER)
-		player.Dead = true
+		player.SetDead()
```
Comment (jwiggins) [inline cart/player.go:160]: Same thing here

**Review by jwiggins — APPROVED** [review]
Thanks for the fixes

**Pull request merged**